### PR TITLE
refact: code refactor

### LINKED ISSUE #82: [REFACT] 코드 모듈화
## Background
- 한 파일안에 있는 함수 및 클래스들을 폴더 별로 분리하여 정리
    
## Todo
- [ ] code refactor

### PR DESCRIPTION
**Pull request by yunguks**
#82

## Overview
- 코드 정리
    
## Change Log
- models, optimizers, datasets, utils 폴더에 각각의 역할마다 파일 분리
    
## To Reviewer    
- 추가적으로 수정하고 싶은 구조 말씀해주세요
    
## Issue Tags
- close #82

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -1,103 +1,21 @@
...
-        matches = [re.search(rf"%s(\d+)" % path.stem, d) for d in dirs]
-        i = [int(m.groups()[0]) for m in matches if m]
-        n = max(i) + 1 if i else 2
-        return f"{path}{n}"
+from datasets.my_dataset import make_dataloader
+from optimizers.loss import create_criterion
+from optimizers.optims import create_optimizer
+from optimizers.schedulers import create_scheduler
+from utils.util import seed_everything, increment_path, get_lr,get_numclass, make_cutmix_input
+
```
Comment (daehun1102) [inline train.py:18]: 너무 깔끔해졌네요

File: inference.py [hunk 2]
```diff
@@ -9,49 +9,8 @@
...
-    # tarpath = os.path.join(saved_model, 'best.tar.gz')
-    # tar = tarfile.open(tarpath, 'r:gz')
-    # tar.extractall(path=saved_model)
-    model_path = os.path.join(saved_model, 'best.pth')
-    model.load_state_dict(torch.load(model_path, map_location=device))
-
-    return model
-
+from datasets.my_dataset import CustomDataset
+from utils.util import read_json, load_model, get_numclass
```
Comment (daehun1102) [inline inference.py:13]: Utils 로 묶으니까 보기좋군요

**Review by daehun1102 — COMMENTED** [review]
수고하셨습니다👍👍

File: datasets/augmentation.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
+import torch
+from torchvision.transforms import Resize, ToTensor, Normalize, Compose, CenterCrop,\
+ColorJitter, RandomRotation, RandomHorizontalFlip
+from ._util import MaskLabels, GenderLabels, AgeLabels
```
Comment (seoin0110) [inline datasets/augmentation.py:4]: 이 부분은 상대 경로로 작성해도 문제가 없을까요?

File: datasets/my_dataset.py [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+from torch.utils.data import Dataset, Subset
+
+from importlib import import_module
+
+from collections import Counter
+import multiprocessing
+
+from torch.utils.data import DataLoader, WeightedRandomSampler
+from sklearn.model_selection import StratifiedKFold
+from datasets._util import GenderLabels, AgeLabels, FILE_NAME
```
Comment (seoin0110) [inline datasets/my_dataset.py:16]: 여기는 augmentation.py나 base_dataset.py와 달리 절대 경로를 사용했네요.

File: datasets/base_dataset.py [hunk 5]
```diff
@@ -0,0 +1,343 @@
+import os
+import random
+from collections import defaultdict
+
+from typing import Tuple, List
+
+import numpy as np
+from PIL import Image
+from torch.utils.data import Dataset, Subset, random_split
+from ._util import MaskLabels, GenderLabels, AgeLabels, FILE_NAME
```
Comment (seoin0110) [inline datasets/base_dataset.py:10]: 이 부분도 확인해주세요!

**Review by seoin0110 — COMMENTED** [review]
깔끔하게 정리되니 보기 좋네요 👍

File: optimizers/loss.py [hunk 6]
```diff
@@ -158,3 +155,18 @@ def create_criterion(criterion_name, **kwargs):
     else:
         raise RuntimeError('Unknown loss (%s)' % criterion_name)
     return criterion
+
+CRITERION_ENTRYPOINTS = {
```
Comment (yunguks) [inline optimizers/loss.py]: 여기 중복 되는 게 있어서 제가 고치고 올릴게요!

**Comment by dh3211** [comment]
LGTM

File: datasets/augmentation.py (hunk 3 above)
Comment (yunguks) [reply datasets/augmentation.py:4]: 동일 폴더 안에 util을 불러오는 거라 상관없을 거 같아요

File: datasets/my_dataset.py (hunk 4 above)
Comment (yunguks) [reply datasets/my_dataset.py:16]: 절대 경로로 통일해서 알아보기 쉽도록 할게요

File: datasets/my_dataset.py [hunk 7]
```diff
@@ -85,203 +86,218 @@ def __getitem__(self, idx):
...
-        if args.category=="mask":
-            for profile in profiles:
-                # img_folder == inputs/train/image/000004_male_Asian_54 
+    train_dataloader.append(train_loader)
+    val_dataloader.append(val_loader)
+    
+    return train_dataloader, val_dataloader
+
+
+def make_kfold_dataset(data_dir,train_transform,val_transform,collate_fn,args):
```
Comment (daehun1102) [inline datasets/my_dataset.py:173]: kfold_dataset 을 따로둬서 너무 깔끔해졌네요

**Review by daehun1102 — COMMENTED** [review]
모듈별로 역할을 나눠서 코드효율이 더 좋아진것같아요👍👍
